Automattic-Tracks-iOSTests/Test Helpers/Mocks.swift

```diff
@@ -0,0 +1,107 @@
+import Foundation
+import Sodium
+@testable import AutomatticTracks
+
+typealias LogFileCallback = (LogFile) -> ()
+typealias ErrorWithLogFileCallback = (Error, LogFile) -> ()
+
+enum MockError: Error {
+    case generic
+}
+
+class MockEventLoggingDataSource: EventLoggingDataSource {
```

Comment:
I think the way this class is written is totally fine for tests, but here is how I would have written it myself: 872326118e4c1820192700f79a62d3f3de162bab.

The idea is to promote immutability and control how this class can be used. It has 0 value in this PR in practice, but I thought I'd take the time to mention this in a couple places in this PR for future reference, only if you find it useful of course.